src/Site.js

```diff
@@ -667,11 +667,15 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
 /**
  * Uses heading data in built pages to generate heading and keyword information for siteData
  * Subsequently writes to siteData.json
+ * @param generateForAllPages if all pages heading data should be generated
+ * @param filePaths optional array of updated file paths during live preview
  */
-Site.prototype.generateSiteData = function () {
+Site.prototype.updateSiteData = function (generateForAllPages = true, filePaths = []) {
```

Comment:
They are mutually exclusive when used right? How about accepting only `filePaths` with no default value, and then assigning `const updateForAllPages = filePaths === undefined;` internally?

Then calling it would be `updateSiteData(shouldRebuildAllPages ? undefined : filePaths)`.

Or pass and check `true` instead of `undefined`.

Reply:
I thought about this but I was afraid that `updateSiteData(shouldRebuildAllPages ? undefined : filePaths)` would be confusing for other devs. I've no objections to it though. I think my main goal was to keep the function call argumentless (ie unchanged) in build process.